fastar/rules.py

```diff
@@ -12,6 +12,22 @@
 map = safe_map
 zip = safe_zip
 
+@curry
+def default_update_rule(backward_rule, fun, cache, outbox, *args, **kwargs):
+  out_start, out_shape = outbox
+  instarts, incounts = backward_rule(outbox, *args, **kwargs)
```

Comment:
In future we might want to make this more efficient, because we don't really need the value of incounts here, just its shape. For now though this is fine.